src/TemplateHelpers.hs

```diff
@@ -125,7 +128,18 @@ renderReadme = \case
 renderHtmlDocs :: D.VerifiedPackage -> Text -> Handler (Maybe Html)
 renderHtmlDocs pkg mnString = do
   htmlRenderContext <- getHtmlRenderContext
-  let docsOutput = packageAsHtml (htmlRenderContext pkg) pkg
+
+  depHtmlRenderContexts <- do
+    Map.fromList . catMaybes <$> do
+      for (D.pkgResolvedDependencies pkg) $ \(pkgName, version) -> do
+        fmap (pkgName, ) <$> do
+          either (const Nothing) Just <$> do
+            fmap htmlRenderContext <$> do
+                lookupPackage pkgName version
+
+  let docsOutput = flip packageAsHtml pkg $ \case
+        D.Local m -> Just $ htmlRenderContext pkg m
+        D.FromDep pkgName m -> return m <**> Map.lookup pkgName depHtmlRenderContexts
```

Comment:
I looked for a function on hoogle `a -> f (a -> b) -> f b`, but only could find `f a -> f (a -> b) -> f b`, hence the `return m` here. Do you know of such a function? Maybe I missed it.